frontends/ast/ast.cc

```diff
@@ -690,12 +694,30 @@ void AstNode::dumpVlog(FILE *f, std::string indent) const
 		break;
 
 	case AST_CAST_SIZE:
-		children[0]->dumpVlog(f, "");
-		fprintf(f, "'(");
-		children[1]->dumpVlog(f, "");
-		fprintf(f, ")");
+		switch (children[0]->type)
+		{
+		case AST_WIRE:
+			if (children[0]->children.size() > 0) {
+				children[0]->children[0]->dumpVlog(f, "");
+				fprintf(f, "'(");
+				children[1]->dumpVlog(f, "");
+				fprintf(f, ")");
+			} else {
+				fprintf(f, "%d'(", children[0]->range_left - children[0]->range_right +1);
+				children[1]->dumpVlog(f, "");
+				fprintf(f, ")");
+			}
+			break;
+
+		default: //AST_IDENTIFIER
+			children[0]->dumpVlog(f, "");
+			fprintf(f, "'(");
+			children[1]->dumpVlog(f, "");
+			fprintf(f, ")");
```

Comment:
Arguably these three lines are now repeated three times. Can we do this once at the end instead?